### DISCUSSION TITLE
Autentification at Microsoft has failed

### DISCUSSION THREAD

**Discussion by Palo47** [lavinir/hassio-onedrive-backup#26] [General]
When I authenticated with Microsoft, the authentication ended with "Your code has expired." When I tried to get a new authentication, it was unsuccessful again.
I then uninstalled and reinstalled One drive Backup but then I could not find the line in the LOG:
 To sign in, use a web browser to open the page https://microsoft.com/deviceloginand enter the code ********* to authenticate. 
There was just a note: 
Warning: token cache is empty
How to fix the issue?

**Comment by lavinir** [comment]
I don't remember what the exact timeout expiration is for the authentication code (maybe ~20min) but it should ask again.
If you uninstalled and re-installed you should certainly see the log again - should be the one right after the **token cache is empty**.
What do you see after that ? Is the addon still running ?

**Reply by Palo47** [reply]
Thank you for your reply. I have been waiting to respond as the issue develops. 

Unfortunately I apparently made a mistake when I reacted to the error response from Microsoft Authenticator unreasonably quickly. It was probably a false error message because Microsoft Authenticator sent an email that granted access for the hassio-one drive-back in the link I see it.

I unfortunately uninstalled AddOn OneDrive Backup from my HA and then reinstalled it. Unfortunately, only the info "token casche is emty" appeared in the log and the new authentication could not be performed
OddOne Onedrive Backup opens in my HA, everything is accessible except that it does not perform any backup to the address in OnDrive.

Do you have any workaround?

**Reply by lavinir** [reply]
Could you clarify please ? You've uninstalled the addon and re-installed but do not get an authentication code past the line of "token cache is empty" ?

**Reply by Palo47** [reply]
So success today.
Circa three days ago I uninstalled OneDrive Backup.
Today circa 30 minutes ago I installed OneDriveBackup again and everything went fine. Also in Ondrive the directory for the backup was created.

I have only been in HA for a short time so I am taking this opportunity to solve my next problem.
Unfortunately I installed the 32 bit version on the RPI 8 GB at the beginning. Shortly after that I wanted to switch to 64 bit but after a new clean install on a new SD card I logged in to an account identical to the old 32 bit install but a cyb message appeared - problem with authentication server (probably for AddOn) and no AddOn could be installed in the install.

I'm putting my hope in a subtopic in which I use a backup of the 32 bit version to transfer the 32 bit version to a not fully functional 64 bit installation.
What procedure do you recommend?

**Reply by Palo47** [reply]
A few days ago I did a fresh install of HA 64 bit on SSD and with a new account.
I successfully installed One Drive Backup on a clean install of HA 64 bit (I did not do a restore from backup).
For a short time One Drive Backup worked properly. Then it stopped working the status was as after the "Stop" action. The attempt to "start" was always unsuccessful, always the status went to "Stop". In the upper right corner of the OneDrive Backup window, there was a consistently prominent red dot.

Uninstalling/installing did not help.

Today I did a fresh clean install of HA 64 bit on the SSD again under another new account (fourth in order) and installed One Drive Backup. Unfortunately it was not possible to authenticate because there was only a message in the log: "Token cashe is emty". Uninstall/Install did not help
It's the same situation as I described in my email of 4 Dec 2022.

Can you help?

**Reply by lavinir** [reply]
The extension actually stops after that log saying the token cache is empty ?
You are on 1.4 now ? Do you have any File Sync Paths defined ?

**Comment by Tonsen01** [comment]
I also get the message, that the "code has expired". I have tried it several times in between seconds.

**Comment by Tonsen01** [comment]
I have just tried it again and it works now. Thanks!
